### PR TITLE
feat: WIP reaction panel component.

### PR DESCRIPTION
**Pull request by AlbaLopez**

### CODE REVIEW

File: src/app/app-interactive-map/components/app-reaction/app-reaction.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import {Component, OnInit} from '@angular/core';
+import {Reaction, Reactions} from '../../../../types/custom_types';
+
+@Component({
+  selector: 'app-reaction',
+  templateUrl: './app-reaction.component.html',
+  styleUrls: ['./app-reaction.component.scss'],
+})
+export class AppReactionComponent implements OnInit {
+  public itemsSelected: Reactions = {
```
Comment (AlbaLopez) [inline src/app/app-interactive-map/components/app-reaction/app-reaction.component.ts]: To separate the panels, I created this object with the differents panels, and then, when you add something, is added to the panel that you choose.

If you are in added reactions, the reactions will be added to itemsSelected added.

File: src/types/custom_types.d.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export interface Reaction {
```
Comment (AlbaLopez) [inline src/types/custom_types.d.ts]: I think we will have to change if we added the bounds etc.

File: src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.scss [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mat-list-item {
+  display: block;
+}
+span {
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.scss]: please add blank lines between definitions

File: src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.scss [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  width: 30px;
+}
+
+.item-display-box {
+  font-size:12px;
+  width: 100px;
+  text-align: center;
+}
+
+.button-apply {
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.scss]: Please try to extract the common parts of `.button-apply` and `.button-reset`.

File: src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.scss [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+.item-display {
+  font-size: 12px;
+  margin-left: 6px;
+  margin-right: 6px;
+  text-align: center !important;
+}
+
+.box-bounds {
+  border:1px solid black;
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.scss]: missing space after `:`

File: src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+})
+export class AppReactionPanelChangedComponent implements OnInit, OnChanges {
+  @Input() public itemsSelected: Reaction[] = [];
+  @Input() public type: string;
+  public clickedItem: string;
+  public lowerbound: number;
+  public upperbound: number;
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-changed/app-reaction-panel-changed.component.ts]: Is this needed?

File: src/app/app-interactive-map/components/app-reaction-panel-detail/app-reaction-panel-detail.component.scss [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-detail/app-reaction-panel-detail.component.scss:1]: Do we need this file?

File: src/app/app-interactive-map/components/app-reaction-panel-detail/app-reaction-panel-detail.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AppReactionPanelDetailComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-detail/app-reaction-panel-detail.component.spec.ts:37]: Please only add tests if you actually intend to add them.

File: src/app/app-interactive-map/components/app-reaction-panel-objective/app-reaction-panel-objective.component.html [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<mat-list *ngIf="item">
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-objective/app-reaction-panel-objective.component.html]: missing license

File: src/app/app-interactive-map/components/app-reaction-panel-objective/app-reaction-panel-objective.component.scss [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+.item-name{
+  width: 100%;
+  margin-left: 8%;
+  display: block;
+}
+mat-slide-toggle {
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-objective/app-reaction-panel-objective.component.scss]: missing balnk line

File: src/app/app-interactive-map/components/app-reaction-panel-objective/app-reaction-panel-objective.component.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+@Component({
+  selector: 'app-reaction-panel-objective',
+  templateUrl: './app-reaction-panel-objective.component.html',
+  styleUrls: ['./app-reaction-panel-objective.component.scss'],
+})
+export class AppReactionPanelObjectiveComponent implements OnInit {
+  @Input() public item;
+
+  constructor() { }
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel-objective/app-reaction-panel-objective.component.ts]: please remove the unused methods if you're not planning to keep them.

File: src/app/app-interactive-map/components/app-reaction-panel/app-reaction-panel.component.scss [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@import '~@angular/material/theming';
+
+mat-card {
+  margin: 5%;
+  box-shadow: 0 1px 3px 0 rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 2px 1px -1px rgba(0,0,0,.12);
+}
+.item-name{
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel/app-reaction-panel.component.scss]: please leave blank lines between blocks

File: src/app/app-interactive-map/components/app-reaction-panel/app-reaction-panel.component.ts [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  displayFn(item: Reaction): string {
+    if (item) {
+      return item.bigg_id;
+    }
+  }
+
+  addItem(reaction: Reaction): void {
+    if (this.type === 'objective') {
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction-panel/app-reaction-panel.component.ts]: Here you could create an object:
```
const typeToTarget = {
  'objective': 'objectiveReaction',
  'added': 'addedReactions',
  'removed': 'removedReactions',
};
// Let's not add cardId, we can grab that in the effect!
if (['added', 'removed'].includes(this.type)) {
  this.store.dispatch(new OperationReaction({reactionId: reaction.biggId,  operationTarget: typeToTarget[this.type]}));
} else {
  this.store.dispatch(new SetObjectiveReaction({reactionId: reaction.bigg_id}));
}
```

File: src/app/app-interactive-map/components/app-reaction/app-reaction.component.scss [hunk 14]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction/app-reaction.component.scss:1]: Please remove if not used

File: src/app/app-interactive-map/components/app-reaction/app-reaction.component.spec.ts [hunk 15]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
```
Comment (matyasfodor) [inline src/app/app-interactive-map/components/app-reaction/app-reaction.component.spec.ts:1]: Please remove if not used

File: src/app/app-interactive-map/components/app-reaction/app-reaction.component.ts (hunk 1 above)
Comment (matyasfodor) [reply src/app/app-interactive-map/components/app-reaction/app-reaction.component.ts]: I don't get why you need this. Can you connect the store?

Comment (AlbaLopez) [reply src/app/app-interactive-map/components/app-reaction/app-reaction.component.ts]: It was three days ago, now I know how is the store working and I have to change everything haha

**Pull request closed without merging**